pre_commit/meta_hooks/check_files_matches_any.py

```diff
@@ -0,0 +1,36 @@
+import re
+import sys
+
+import pre_commit.constants as C
+from pre_commit.clientlib import load_config
+from pre_commit.git import get_all_files
+
+
+def files_matches_any(filenames, include):
+    include_re = re.compile(include)
+    for filename in filenames:
+        if include_re.search(filename):
+            return True
+    return False
+
+
+def check_files_matches_any(config_file=None):
+    config = load_config(config_file or C.CONFIG_FILE)
+    files = get_all_files()
+    files_not_matched = False
+
+    for repo in config['repos']:
+        for hook in repo['hooks']:
+            include = hook.get('files', '')
```

Comment:
`files` will always be present when loaded through `load_config` due to `pre_commit.schema` -- this can be `hook['files']`

Reply:
`load_config` does not merge the data from `.pre-commit-config.yaml` with the configuration in the hook repo, that happens in `Repository.hooks`.

Reply:
oh true!  totally forgot about this 👍